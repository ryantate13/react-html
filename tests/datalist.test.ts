import {datalist} from '../src';

describe('datalist', () => {
    it('renders with children', () => {
        expect(datalist({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(datalist({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(datalist()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(datalist({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(datalist({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(datalist({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(datalist({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(datalist({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(datalist({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(datalist({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(datalist({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(datalist({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(datalist({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(datalist({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(datalist({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(datalist({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(datalist({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(datalist({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(datalist({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(datalist({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(datalist({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(datalist({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(datalist({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(datalist({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(datalist({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(datalist({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(datalist({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(datalist({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(datalist({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(datalist({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(datalist({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(datalist({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(datalist({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(datalist({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(datalist({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(datalist({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(datalist({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(datalist({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(datalist({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(datalist({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(datalist({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(datalist({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(datalist({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(datalist({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(datalist({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(datalist({translate: "no"})).toMatchSnapshot();
    });
});