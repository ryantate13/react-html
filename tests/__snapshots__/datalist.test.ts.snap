// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datalist renders arrays of children 1`] = `
<datalist>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</datalist>
`;

exports[`datalist renders with accessKey="a,b,c,d" 1`] = `
<datalist
  accessKey="a,b,c,d"
/>
`;

exports[`datalist renders with autoCapitalize="characters" 1`] = `
<datalist
  autoCapitalize="characters"
/>
`;

exports[`datalist renders with autoCapitalize="none" 1`] = `
<datalist
  autoCapitalize="none"
/>
`;

exports[`datalist renders with autoCapitalize="off" 1`] = `
<datalist
  autoCapitalize="off"
/>
`;

exports[`datalist renders with autoCapitalize="on" 1`] = `
<datalist
  autoCapitalize="on"
/>
`;

exports[`datalist renders with autoCapitalize="sentences" 1`] = `
<datalist
  autoCapitalize="sentences"
/>
`;

exports[`datalist renders with autoCapitalize="words" 1`] = `
<datalist
  autoCapitalize="words"
/>
`;

exports[`datalist renders with children 1`] = `
<datalist>
  TEST
</datalist>
`;

exports[`datalist renders with className="foo bar baz" 1`] = `
<datalist
  className="foo bar baz"
/>
`;

exports[`datalist renders with contentEditable="false" 1`] = `
<datalist
  contentEditable="false"
/>
`;

exports[`datalist renders with contentEditable="true" 1`] = `
<datalist
  contentEditable="true"
/>
`;

exports[`datalist renders with dir="auto" 1`] = `
<datalist
  dir="auto"
/>
`;

exports[`datalist renders with dir="ltr" 1`] = `
<datalist
  dir="ltr"
/>
`;

exports[`datalist renders with dir="rtl" 1`] = `
<datalist
  dir="rtl"
/>
`;

exports[`datalist renders with draggable="false" 1`] = `
<datalist
  draggable="false"
/>
`;

exports[`datalist renders with draggable="true" 1`] = `
<datalist
  draggable="true"
/>
`;

exports[`datalist renders with empty attributes 1`] = `<datalist />`;

exports[`datalist renders with hidden=false 1`] = `
<datalist
  hidden={false}
/>
`;

exports[`datalist renders with hidden=true 1`] = `
<datalist
  hidden={true}
/>
`;

exports[`datalist renders with id="TEST" 1`] = `
<datalist
  id="TEST"
/>
`;

exports[`datalist renders with inputMode="decimal" 1`] = `
<datalist
  inputMode="decimal"
/>
`;

exports[`datalist renders with inputMode="email" 1`] = `
<datalist
  inputMode="email"
/>
`;

exports[`datalist renders with inputMode="none" 1`] = `
<datalist
  inputMode="none"
/>
`;

exports[`datalist renders with inputMode="numeric" 1`] = `
<datalist
  inputMode="numeric"
/>
`;

exports[`datalist renders with inputMode="search" 1`] = `
<datalist
  inputMode="search"
/>
`;

exports[`datalist renders with inputMode="tel" 1`] = `
<datalist
  inputMode="tel"
/>
`;

exports[`datalist renders with inputMode="text" 1`] = `
<datalist
  inputMode="text"
/>
`;

exports[`datalist renders with inputMode="url" 1`] = `
<datalist
  inputMode="url"
/>
`;

exports[`datalist renders with is="foo" 1`] = `
<datalist
  is="foo"
/>
`;

exports[`datalist renders with itemID="https://google.com/" 1`] = `
<datalist
  itemID="https://google.com/"
/>
`;

exports[`datalist renders with itemProp="https://google.com/" 1`] = `
<datalist
  itemProp="https://google.com/"
/>
`;

exports[`datalist renders with itemRef="foo bar" 1`] = `
<datalist
  itemRef="foo bar"
/>
`;

exports[`datalist renders with itemScope=false 1`] = `
<datalist
  itemScope={false}
/>
`;

exports[`datalist renders with itemScope=true 1`] = `
<datalist
  itemScope={true}
/>
`;

exports[`datalist renders with itemType="https://google.com/" 1`] = `
<datalist
  itemType="https://google.com/"
/>
`;

exports[`datalist renders with lang="" 1`] = `
<datalist
  lang=""
/>
`;

exports[`datalist renders with lang="es-419" 1`] = `
<datalist
  lang="es-419"
/>
`;

exports[`datalist renders with slot="TEST" 1`] = `
<datalist
  slot="TEST"
/>
`;

exports[`datalist renders with spellCheck="false" 1`] = `
<datalist
  spellCheck="false"
/>
`;

exports[`datalist renders with spellCheck="true" 1`] = `
<datalist
  spellCheck="true"
/>
`;

exports[`datalist renders with style={"color":"#000000"} 1`] = `
<datalist
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`datalist renders with tabIndex=1 1`] = `
<datalist
  tabIndex={1}
/>
`;

exports[`datalist renders with title="TEST" 1`] = `
<datalist
  title="TEST"
/>
`;

exports[`datalist renders with translate="no" 1`] = `
<datalist
  translate="no"
/>
`;

exports[`datalist renders with translate="yes" 1`] = `
<datalist
  translate="yes"
/>
`;

exports[`datalist renders without attributes 1`] = `<datalist />`;
